controllers/csm_controller.go

```diff
@@ -694,17 +698,17 @@ func (r *ContainerStorageModuleReconciler) removeDriver(ctx context.Context, ins
 
 // PreChecks - validate input values
 func (r *ContainerStorageModuleReconciler) PreChecks(ctx context.Context, cr *csmv1.ContainerStorageModule, operatorConfig utils.OperatorConfig) error {
-	if cr.Spec.Driver.ConfigVersion == "" || cr.Spec.Driver.ConfigVersion != "v2.2.0" {
-		return fmt.Errorf("driver version not specified in spec or driver version is not supported")
-	}
-
 	// Check drivers
 	switch cr.Spec.Driver.CSIDriverType {
 	case csmv1.PowerScale:
 		err := drivers.PrecheckPowerScale(ctx, cr, r.GetClient())
 		if err != nil {
 			return fmt.Errorf("failed powerscale validation: %v", err)
 		}
+		if cr.Spec.Driver.ConfigVersion == "" || !(PScaleSupportedDriverVersions[cr.Spec.Driver.ConfigVersion]) {
```

Comment:
Please move this version check to https://github.com/dell/csm-operator/blob/7f1d2c2c84c7c6f6a88aaf2e3a134f58759fa565/pkg/drivers/powerscale.go#L27

All module/driver specific checks should be handled in their respective pkg

Reply:
@medegw01 done